test/metabase/api/downloads_exports_test.clj

```diff
@@ -0,0 +1,184 @@
+                                              :source-table (mt/id :pivot-base)}}}]
+      (let [result (->> (mt/user-http-request :crowberto :post 200 (format "card/%d/query/csv?format_rows=false" card-id))
+                        csv/read-csv)]
+        (testing "Pivot CSV Exports look like a Pivoted Table"
+          (testing "The Headers Properly indicate the pivot rows names."
+            ;; Pivot Rows Header are Simply the Column names from the rows specified in
+            ;; [:visualization_settings :pivot_table.column_split :rows]
+            (is (= [["Cat C" "Cat D"]
+                    ["Cat C" "Cat D"]]
+                   [(take 2 (first result))
+                    (take 2 (second result))])))
+          (testing "The Headers Properly indicate the pivot column names."
+            (let [[header1 header2] (map set (take 2 result))
+                  possible-vals-of-a #{"AA" "AB" "AC" "AD"}
+                  possible-vals-of-b #{"BA" "BB" "BC" "BD"}]
+              ;; In a Pivot Table, the Column headers are derived from the Possible Values in each Pivot Column
+              ;; Since the test data used here has defined 2 Pivot Columns, there must be 2 Header rows
+              ;; to fully capture all of the combinations of possible values for each pivot-col specified in
+              ;; [:visualization_settings :pivot_table.column_split :columns]
+              ;; To hopefully illustrate a bit, Let's consider that:
+              ;; Cat A can have "AA" "AB" "AC" "AD"
+              ;; Cat B can have "BA" "BB" "BC" "BD"
+              ;; The first 4 Entries in Header 1 (excluding the Pivot Rows Display Names) will all be "AA"
+              ;; And the first 4 Entries in Header 2 will be "BA" "BB" "BC" "BD"
+              (is (= [["AA" "AA" "AA" "AA"]
+                      ["BA" "BB" "BC" "BD"]]
+                     [(take 4 (drop 2 (first result)))
+                      (take 4 (drop 2 (second result)))]))
+              ;; This combination logic would continue for each specified Pivot Column, but we'll just stick with testing 2
+              ;; To keep things relatively easy to read and understand.
+              (testing "The first Header only contains possible values from the first specified pivot column"
+                (is (set/subset? possible-vals-of-a header1))
+                (is (not (set/subset? possible-vals-of-b header1))))
+              (testing "The second Header only contains possible values from the second specified pivot column"
+                (is (set/subset? possible-vals-of-b header2))
+                (is (not (set/subset? possible-vals-of-a header2))))
+              (testing "The Headers also show the Row Totals header"
+                (is (= ["Row totals"
+                        "Row totals"]
+                       (map last (take 2 result))))))))
+
+        (testing "The Columns Properly indicate the pivot row names."
```

Comment:
Generally any time you're putting blank lines in big tests to break it up you might want to consider seeing if there's an easy way to break it up into several smaller tests. Otherwise these tests that assert 100 things get really hard to debug. I guess this one is ok for now since this isn't super easy to break into two tests unless you want to run that query twice

Reply:
I think I'll fix the test up more in a follow up PR.